### PR TITLE
Feature messaging

### PR DESCRIPTION
**Pull request by sholabioku**
### What this PR does (required):

--It creates a new conversation
--It sends message
--It gets all conversations
--It gets all messages for a conversation

### Screenshots / Videos (front-end only):

- Post at least one screenshot to show the feature in action. **Only required in front-end work**

### Any information needed to test this feature (required):

- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):

- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: server/controllers/conversation.js
Comment (ugur-akin) [inline server/controllers/conversation.js:32]: I think should send both responses in a consistent manner (e.g. this line and `line 44`).

File: server/controllers/message.js
Comment (ugur-akin) [inline server/controllers/message.js:35]: This is nice for now, but we will probably need to modify it later to use `socket.io` 👍🏼

**Review by ugur-akin — COMMENTED** [review]
This looks great, good job @sholabioku ❗ 💯 🎉 

I'm mostly okay with everything here but I'm not the most experienced when it comes to messaging so let's wait for @rajivtitus's approval before we merge this in 👍🏼

File: server/models/Message.js
Comment (rajivtitus) [inline server/models/Message.js:18]: 👍🏽

File: server/models/Conversation.js
Comment (rajivtitus) [inline server/models/Conversation.js:14]: 👍🏽

File: server/controllers/conversation.js
Comment (rajivtitus) [inline server/controllers/conversation.js:9]:
```suggestion
  const { receiverId } = req.body;
```
To avoid any confusion we can change the above to `receiverId`, as `userId` in some cases would refer to the logged in user 😄

File: server/controllers/message.js
Comment (rajivtitus) [inline server/controllers/message.js:10]: Optional: You can perform path parameter validation and return a `400` if it is missing

**Review by rajivtitus — APPROVED** [review]
LGTM! Nice work with those changes 💯

**Pull request merged**
